ui/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/ManagementActor.scala

```diff
@@ -118,28 +118,52 @@ class ManagementActor(managers: Map[ProcessingType, ProcessManager],
 
   //This method handles some corner cases like retention for keeping old states - some engine can cleanup canceled states. It's more Flink hermetic.
   //TODO: In future we should move this functionality to ProcessManager.
-  private def handleObsoleteStatus(processStateDefinitionManager: ProcessStateDefinitionManager, processState: Option[ProcessState], lastAction: Option[ProcessAction]): ProcessStatus =
+  private def handleObsoleteStatus(processState: Option[ProcessState], lastAction: Option[ProcessAction]): ProcessStatus =
     (processState, lastAction) match {
-      case (Some(state), _)  => handleMismatchStateVersion(state, lastAction)
-      case (None, Some(act)) if act.isCanceled => ProcessStatus.simple(SimpleStateStatus.Canceled)
-      case (None, Some(act)) if act.isDeployed => ProcessStatus.simpleErrorShouldRunning(act.processVersionId, act.user, Option.empty)
+      case (_, Some(action)) if action.isDeployed => handleMismatchDeployedLastAction(processState, action)
+      case (Some(state), _) if state.status.isFollowingDeployAction => handleFollowingDeployState(state, lastAction)
+      case (None, Some(action)) if action.isCanceled => ProcessStatus.simple(SimpleStateStatus.Canceled)
+      case (Some(state), _) => ProcessStatus(state)
       case (None, None) => ProcessStatus.simple(SimpleStateStatus.NotDeployed)
     }
 
-  //This method handles some corner cases for mismatch states version
+  //This method handles some corner cases for following deploy state mismatch last action version
   //TODO: In future we should move this functionality to ProcessManager.
-  private def handleMismatchStateVersion(processState: ProcessState, lastAction: Option[ProcessAction]): ProcessStatus = {
-    (processState.version, lastAction) match {
-      case (Some(sv), Some(act)) if act.isDeployed && sv.versionId != act.processVersionId =>
-        ProcessStatus.simpleErrorMismatchDeployedVersion(act.processVersionId, sv.versionId, act.user, Some(processState))
-      case (Some(_), Some(act)) if act.isDeployed && !processState.status.isFollowingDeployAction =>
-        ProcessStatus.simpleErrorShouldRunning(act.processVersionId, act.user, Some(processState))
-      case (Some(version), None) if processState.isDeployed =>
-        ProcessStatus.simpleErrorShouldNotBeDeployed(version.versionId, version.user, Some(processState))
-      case (Some(stateVersion), Some(action)) if stateVersion.versionId == action.processVersionId =>
-        ProcessStatus(processState)
+  private def handleFollowingDeployState(state: ProcessState, lastAction: Option[ProcessAction]): ProcessStatus =
+    lastAction match {
+      case Some(action) if action.isCanceled =>
+        state.version match {
+          case Some(ver) =>
+            ProcessStatus.simpleErrorShouldNotBeDeployed(ver.versionId, ver.user, Option(state))
```

Comment:
can you add here and in other error places some log.warn with message that will help us distinguish those cases which has the same message but in different case handler?

Reply:
I've removed this `state.version match`. Now we have only ` case Some(action) if action.isCanceled =>` in fact these old cases did same things.